crates/bevy_time/src/commands.rs

```diff
@@ -0,0 +1,290 @@
+use std::{
+    any::TypeId,
+    ops::{Deref, DerefMut},
+    time::Duration,
+};
+
+use bevy_ecs::{
+    system::{Commands, Deferred, Res, ResMut, Resource, SystemBuffer, SystemMeta, SystemParam},
+    world::{CommandQueue, World},
+};
+use bevy_utils::TypeIdMap;
+
+use crate::{Domain, Fixed, Generic, Time, Timer, TimerMode, Virtual};
+
+/// A [`SystemParam`] that allows you to queue commands to be
+/// executed after a certain duration has elapsed,
+/// rather than immediately at the next sync point.
+///
+/// This type also derefs to [`Commands`], so you can still use it
+/// to queue commands to run at the next sync point.
+///
+/// # Deferred
+///
+/// - Commands queued during the [`Update`] schedules
+///   will be applied (if their timers have elapsed) during the [`First`] schedule.
+/// - Commands queued during the [`FixedUpdate`] schedules
+///   will be applied (if their timers have elapsed) during the [`FixedFirst`] schedule.
```

Comment:
Shouldn't the commands still be applied in the First schedule? If I queue something to run after 5 seconds, I would expect this to run as soon as the timer elapsed 5 seconds